### PR TITLE
Nazanin/promp_trade

### PR DESCRIPTION
**Pull request by nazaninreihani**

### CODE REVIEW

File: src/javascript/app/pages/trade/purchase.js [hunk 1]
```diff
@@ -62,23 +64,31 @@ const Purchase = (() => {
...
-            confirmation_error.show();
-            let message = error.message;
-            if (/RestrictedCountry/.test(error.code)) {
-                let additional_message = '';
-                if (/FinancialBinaries/.test(error.code)) {
-                    additional_message = localize('Try our [_1]Volatility Indices[_2].', [`<a href="${urlFor('get-started/binary-options', 'anchor=volatility-indices#range-of-markets')}" >`, '</a>']);
-                } else if (/Random/.test(error.code)) {
-                    additional_message = localize('Try our other markets.');
+            if (/AuthorizationRequired/.test(error.code)) {
+                $(authorization_error).setVisibility(true);
```
Comment (negarn) [inline src/javascript/app/pages/trade/purchase.js]: generally we use `.setVisibility(1)`

File: src/javascript/app/pages/trade/purchase.js [hunk 2]
```diff
@@ -62,23 +64,31 @@ const Purchase = (() => {
...
+                const authorization_error_btn_login = CommonFunctions.getElementById('authorization_error_btn_login');
+                authorization_error_btn_login.removeEventListener('click', loginOnClick);
+                authorization_error_btn_login.addEventListener('click', loginOnClick);
+            } else {
+                $(confirmation_error).setVisibility(true);
```
Comment (negarn) [inline src/javascript/app/pages/trade/purchase.js]: same here

File: src/javascript/app/pages/trade/purchase.js [hunk 3]
```diff
@@ -62,23 +64,31 @@ const Purchase = (() => {
...
+                CommonFunctions.elementInnerHtml(confirmation_error, message);
             }
-            CommonFunctions.elementInnerHtml(confirmation_error, message);
         } else {
             CommonFunctions.getElementById('guideBtn').style.display = 'none';
             container.style.display = 'table-row';
             message_container.show();
-            confirmation_error.hide();
+            $(authorization_error).setVisibility(false);
+            $(confirmation_error).setVisibility(false);
```
Comment (negarn) [inline src/javascript/app/pages/trade/purchase.js]: same here, these should be `.setVisibility(0)`

File: src/javascript/app/pages/trade/purchase.js [hunk 4]
```diff
@@ -17,6 +17,7 @@ const urlFor                   = require('../../../_common/url').urlFor;
 const createElement            = require('../../../_common/utility').createElement;
 const getPropertyValue         = require('../../../_common/utility').getPropertyValue;
 const template                 = require('../../../_common/utility').template;
+const Login                    = require('../../../_common/base/login');
```
Comment (negarn) [inline src/javascript/app/pages/trade/purchase.js]: should be on line 12, alphabetically sorted by path

File: src/javascript/app/pages/trade/purchase.js [hunk 5]
```diff
@@ -62,23 +64,31 @@ const Purchase = (() => {
...
-            confirmation_error.show();
-            let message = error.message;
-            if (/RestrictedCountry/.test(error.code)) {
-                let additional_message = '';
-                if (/FinancialBinaries/.test(error.code)) {
-                    additional_message = localize('Try our [_1]Volatility Indices[_2].', [`<a href="${urlFor('get-started/binary-options', 'anchor=volatility-indices#range-of-markets')}" >`, '</a>']);
-                } else if (/Random/.test(error.code)) {
-                    additional_message = localize('Try our other markets.');
+            if (/AuthorizationRequired/.test(error.code)) {
+                $(authorization_error).setVisibility(1);
```
Comment (negarn) [inline src/javascript/app/pages/trade/purchase.js]: no need for jquery wrapper, we have a setVisibility prototype function declared for DOM

File: src/javascript/app/pages/trade/purchase.js [hunk 6]
```diff
@@ -62,23 +64,31 @@ const Purchase = (() => {
...
+                const authorization_error_btn_login = CommonFunctions.getElementById('authorization_error_btn_login');
+                authorization_error_btn_login.removeEventListener('click', loginOnClick);
+                authorization_error_btn_login.addEventListener('click', loginOnClick);
+            } else {
+                $(confirmation_error).setVisibility(1);
```
Comment (negarn) [inline src/javascript/app/pages/trade/purchase.js]: same here

File: src/templates/app/trade/trading.jsx [hunk 7]
```diff
@@ -210,7 +210,17 @@ const Trading = () => (
...
+                                    <h3 id='authorization_error_text'>{it.L('Ready to trade?')}</h3>
+                                    <a className='button' href={it.url_for('new-account')}>
+                                        <span>{it.L('Open A Free Account')}</span>
+                                    </a>
+                                    <p>{it.L('Already have an account?')}</p>
+                                    <a id='authorization_error_btn_login' href='javascript:;'>{it.L('Log in here')}</a>
+                                </div>
+                            </div>
                         </div>
                         <div id='confirmation_message_endelement' />
```
Comment (zakibinary) [inline src/templates/app/trade/trading.jsx]: Do we still need this element?

**Pull request merged**